src/javascript/gameInstance.js

```diff
@@ -194,63 +349,62 @@ export default class GameInstance {
     const intervalPrePlay = setInterval(() => {
       const elapsedTime = Date.now() - startTime;
       this.playTime = Number(elapsedTime / 1000);
+      this.paused = false;
       if (this.playTime > Number(this.vm.noteSpeedInSec)) {
         if (!this.vm.started || !this.paused) this.resumeGame();
         // this.vm.visualizerInstance.initAllVisualizersIfRequried();
         clearInterval(intervalPrePlay);
         clearInterval(this.intervalPlay);
         this.vm.started = true;
-        this.intervalPlay = setInterval(async () => {
-          const cTime = await this.getCurrentTime();
-          this.playTime = cTime + Number(this.vm.noteSpeedInSec);
-          const lastIdx = this.timeArr.length - 1;
-          // advance time arr idx if it's behind current play time
-          if (this.vm.inEditor && !this.vm.playMode && !this.paused) {
-            const lastIdx = this.timeArrIdx;
-            let idx = this.timeArr.findIndex((e) => e.t >= cTime);
-            if (idx === -1) {
-              const endIdx = this.timeArr.length - 1;
-              idx =
-                this.timeArr[endIdx] && this.timeArr[endIdx].t < cTime
-                  ? endIdx + 1
-                  : 0;
-            }
-            this.timeArrIdx = idx;
-            if (
-              this.vm.showExistingNote &&
-              lastIdx !== idx &&
-              this.timeArr[idx]
-            )
-              this.registerKeyDown(this.timeArr[idx].k, "grey");
-          }
-          // check game end
-          if (
-            !this.vm.inEditor &&
-            this.vm.playMode &&
-            this.timeArrIdx >= lastIdx &&
-            this.timeArr[lastIdx] &&
-            this.playTime > Number(this.timeArr[lastIdx].t) + 5
-          ) {
-            this.vm.gameEnded();
-          }
-        }, 100);
+        this.intervalPlay = setInterval(this.gameTimingLoop.bind(this), 100);
       }
     }, 100);
   }
 
+  async gameTimingLoop() {
+    const cTime = await this.getCurrentTime();
+    this.playTime = cTime + Number(this.vm.noteSpeedInSec);
+    this.currentTime = cTime;
+
+    // check game end
+    const lastIdx = this.timeArr.length - 1;
+    if (
+      !this.vm.inEditor &&
+      this.vm.playMode &&
+      this.timeArrIdx >= lastIdx &&
+      this.timeArr[lastIdx] &&
+      this.playTime > Number(this.timeArr[lastIdx].t) + 5
+    ) {
+      this.vm.gameEnded();
+    }
+  }
+
   getCurrentTime() {
     // it seems that 'getPlayerTime' is async, thus all places calling this func need to await res [help wanted]
     return this.vm.srcMode === "youtube"
       ? this.ytPlayer.getPlayerTime()
       : this.audio.getCurrentTime();
   }
```

Comment:
I think this should be `async` no?

Reply:
Well I guess it's not required here, since `getPlayerTime` will return a promise anyway, otherwise `getCurrentTime` just returns a number.

Reply:
What do you think?

Reply:
Right, you are right. But I wonder would `Promise.resolve(this.audio.getCurrentTime());` be any different from only returning the value in terms of the final result?